src/Dropdown/Dropdown.js

```diff
@@ -17,34 +17,55 @@
 /* @flow */
 import React, { Component } from 'react';
 import { findDOMNode } from 'react-dom';
-import { generateId } from '../utils';
+import scrollIntoViewIfNeeded from 'scroll-into-view-if-needed';
+import { composePropsWithGetter, generateId } from '../utils';
 import Root from '../Popover';
-import DropdownContent from './DropdownContent';
+import DropdownContent, {
+  componentTheme as dropdownContentComponentTheme
+} from './DropdownContent';
+import ItemMatcher from './ItemMatcher';
+import { getItems } from '../Menu/Menu';
 
-type Item = {
-  iconEnd?: React$Element<*>,
-  iconStart?: React$Element<*>,
-  disabled?: boolean,
-  divider?: boolean,
-  onClick?: (event: SyntheticEvent<>) => void,
-  render?: (item: Object, props: Object, theme: Object) => React$Element<*>,
-  secondaryText?: React$Node,
-  text?: React$Node,
-  variant?: 'regular' | 'danger' | 'success' | 'warning'
-};
+import type { Items, ItemGroups } from '../Menu/Menu';
+import type { Item } from '../Menu/MenuItem';
 
 type Props = {
   /** Trigger for the Dropdown */
   children: React$Node,
-  /** Open the Dropdown immediately upon initialization */
+  /**
+   * Open the Dropdown upon initialization. Primarily for use with uncontrolled
+   * components.
+   */
   defaultIsOpen?: boolean,
+  /**
+   * Index of item to be highlighted upon initialization. Primarily for
+   * use with uncontrolled components.
+   */
+  defaultHighlightedIndex?: number,
   /** Disable the Dropdown */
   disabled?: boolean,
-  /** Data from which the [Menu](../menu#data) will be constructed (see [example](#data)) */
-  data: Array<{ items: Array<Item>, title?: React$Node }>,
-  /** For use with controlled components, in which the app manages Dropdown state */
+  /**
+   * Data from which the [Menu](../menu#data) will be constructed
+   * (see [example](#data))
+   */
+  data: Items | ItemGroups,
+  /** @Private Function that returns props to be applied to each item */
+  getItemProps?: (props: Object, scope?: Object) => Object,
+  /** @Private Function that returns props to be applied to the menu */
+  getMenuProps?: (props: Object) => Object,
+  /** @Private Function that returns props to be applied to the trigger */
+  getTriggerProps?: (props: Object) => Object,
+  /** Index of the highlighted item. For use with controlled components. */
+  highlightedIndex?: number,
```

Comment:
It seems inconsistent for `highlightedIndex` to use an index and `selectedItem` to use an Item.